tsl/src/nodes/data_node_copy.c

```diff
@@ -139,6 +150,30 @@ data_node_copy_begin(CustomScanState *node, EState *estate, int eflags)
 
 	ps = ExecInitNode(subplan, estate, eflags);
 
+	switch (nodeTag(ps))
+	{
+		case T_ResultState:
+		{
+			/* The planner injected a Result node so we need to get the
+			 * ChunkDispatchState from the Result's child */
+			const ResultState *result = castNode(ResultState, ps);
+			PlanState *child = result->ps.lefttree;
+
+			if (child != NULL && ts_is_chunk_dispatch_state(child))
+				dncs->cds = (ChunkDispatchState *) child;
+			break;
+		}
+		case T_CustomScanState:
+			if (ts_is_chunk_dispatch_state(ps))
+				dncs->cds = (ChunkDispatchState *) ps;
+			break;
+		default:
+			break;
+	}
+
+	if (NULL == dncs->cds)
+		elog(ERROR, "unexpected child plan node %d for DataNodeCopy", nodeTag(ps));
```

Comment:
No test case for this alternative. Can you add one?

Reply:
No, since I don't know a query that would insert other plan nodes. I added this as a safeguard for future unknown query plans (or node injections) that we haven't handled instead of outright crashing in weird ways.